cmd/set.go

```diff
@@ -46,13 +46,31 @@ func Set(environment string, subcomponent string, pathValuePairStrings []string)
 		return err
 	}
 
+	newConfigError := "New configuration was specified and the --newconfigfail switch is on."		
+	
 	for _, pathValue := range pathValuePairs {
+			if newconfigfail {
+				if (!componentConfig.HasSubcomponentConfig(subcomponentPath)) {
+					return errors.New(newConfigError)
+				} else {
+					sc := componentConfig.GetSubcomponentConfig(subcomponentPath)
+
+					if(!sc.HasComponentConfig(pathValue.Path)) {
+						return errors.New(newConfigError)
```

Comment:
factor out this errors.New to the newConfigError definition and then just return newConfigError